### PR TITLE
Avoid checking file permissions in getFileInfo method

### PR DESCRIPTION
**Pull request by Haoning-Sun**
### What changes are proposed in this pull request?

Fix https://github.com/Alluxio/alluxio/issues/15936

### CODE REVIEW

**Review by yyongycy — COMMENTED** [review]
Please check the hdfs behavior.

**Comment by LuQQiu** [comment]
Also commented in the github issue, thanks!

**Comment by Haoning-Sun** [comment]
I tested the getFileStatus method of HDFS, and only need to have execution permission on the parent folder.

image
image
```
public static void printStatus(FileSystem fileSystem, String path) {
        try {
            FileStatus fileStatus = fileSystem.getFileStatus(new Path(path));
            System.out.println(path + " permission: " + fileStatus.getPermission().toString());
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
```
image

* no permission for parent folder.
image

**Review by yyongycy — APPROVED** [review]
lgtm

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java [hunk 1]
```diff
@@ -897,7 +897,7 @@ public FileInfo getFileInfo(AlluxioURI path, GetStatusContext context)
         try (LockedInodePath inodePath = mInodeTree.lockInodePath(lockingScheme)) {
           auditContext.setSrcInode(inodePath.getInodeOrNull());
           try {
-            mPermissionChecker.checkPermission(Mode.Bits.READ, inodePath);
```
[the remark this answers is not in the document]
Comment (Haoning-Sun) [reply core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java:900]: Both posix and hdfs check the execution permission recursively and do not need the read permission. And user do not need to have permissions on the current file.

[the remark this answers is not in the document]
Comment (yyongycy) [reply core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java:900]: thanks for the investigation.

**Review by dbw9580 — APPROVED** [review]
LGTM

**Comment by yyongycy** [comment]
alluxio-bot, merge this please

**Pull request merged**